packages/core/documentation/src/components/DocumentationContent.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div class="documentation">
+    <DocumentationPageEmptyState
+      v-if="documentList && !documentList.length"
+      :can-edit="canEdit"
+      @create-documentation="handleAddClick"
+    />
+    <KCard
+      v-else
+      border-variant="noBorder"
+      class="documentation-card"
+    >
+      <template #body>
+        <div class="documentation-parent-container">
+          <KTreeList
+            :key="`tree-list-${cacheKey}`"
+            class="document-tree-list"
+            :items="documentList"
+            :max-depth="2"
+            @change="(data) => emit('parent-change', data)"
+            @child-change="(data) => emit('child-change', data)"
+            @selected="(data) => emit('document-selection', data)"
+          />
+          <DocumentationDisplay
+            v-if="selectedDocument"
+            :key="key"
+            :can-edit="canEdit"
+            class="document-holder"
+            :hide-publish-toggle="hidePublishToggle"
+            :is-card="isCard"
+            :selected-document="selectedDocument"
+            @add-clicked="handleAddClick"
+            @edit-clicked="handleEditClick"
+            @toggle-published="(data) => emit('toggle-published', data)"
+          />
+        </div>
+      </template>
+    </KCard>
+    <ProductDocumentModal
+      v-if="displayModal"
+      :documents="documentList"
+      :error-message="modalErrorMessage"
+      :hide-publish-toggle="hidePublishToggle"
+      :is-editing="isEditing"
+      :record="isEditing && selectedDocument ? selectedDocument : undefined"
+      @canceled="handleModalClosed"
+      @delete="emit('delete')"
+      @save="(formData: FormData, selectedFile: any) => emit('save', formData, selectedFile)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+import DocumentationDisplay from './DocumentationDisplay.vue'
+import DocumentationPageEmptyState from './DocumentationPageEmptyState.vue'
+import ProductDocumentModal from './ProductDocumentModal.vue'
+import type { PropType } from 'vue'
+import type { DocumentListItem, DocumentTree, FormData } from '../types'
+import type { ChangeEvent, ChildChangeEvent, TreeListItem } from '@kong/kongponents'
+
+const emit = defineEmits<{
+  (e: 'child-change', data: ChildChangeEvent): void,
+  (e: 'delete'): void,
+  (e: 'document-selection', data: TreeListItem): void,
+  (e: 'modal-closed'): void,
+  (e: 'parent-change', data: ChangeEvent): void,
+  (e: 'save', formData: FormData, selectedFile: any): void,
+  (e: 'toggle-published', data: boolean): void,
+}>()
+
+const displayModal = ref<boolean>(false)
+const isEditing = ref<boolean>(false)
+
+const props = defineProps({
+  actionSuccess: {
+    type: Boolean,
+    default: false,
+  },
+  cacheKey: {
```

Comment:
question: what’s the purpose this cache key prop?

Reply:
It was part of the original existing docs code that I did not modify in attempt not to break things during the port. I assume it's for this: https://vuejs.org/api/built-in-special-attributes.html#key

Reply:
I'm pretty sure it's used in the parent component to refresh the tree list component when an update event happens in the parent. This prop gets folded into the `key` attribute of the `KTreeList` instance up above.